pylint/checkers/variables.py

```diff
@@ -1351,6 +1352,22 @@ def visit_global(self, node: nodes.Global) -> None:
             self.add_message("global-statement", node=node)
 
     def visit_assignname(self, node: nodes.AssignName) -> None:
+        if self.linter.is_message_enabled("redefined-outer-name") and isinstance(
```

Comment:
Is `self.linter.is_message_enabled` needed? This tends to interact badly with `useless-suppression` as message don't get flagged as being normally emitted.

Reply:
Wow, thanks for teaching me about that. I must have picked this up from other examples. That's an audit opportunity. I guess it's used for performance in some places but at the expense of false negatives for `useless-suppression`?

Reply:
To add to what @DanielNoord said, I think the most "bang for the buck" issue regarding performance is the startup time ([importing astroid before launching pylint](https://github.com/PyCQA/astroid/issues/1320)). Sure the inference is what takes the most time overall. But I think it's rare that pylint is run on an entire code base by a user, it's done in CI where responsiveness is less of an issue. When a user launch pylint on a single file or on a small number of files he or she just modified then the startup time is taking more than 50% of the total time and make pylint feel very slow.